### DISCUSSION TITLE
How to create with optional association?

### DISCUSSION THREAD

**Discussion by jeancarlos-cpu** [prisma/prisma#2608] [Q&A] [answered]
Hello! I am trying to create a user with an optional address and below is my solution.

I ask if there are any best practices for doing this. I have another entity with 4 optional associations and it will be very messy if I do it like this.

```
createUser({ user, address = {} }) {
    const { cityId, stateId } = address;
    delete address.cityId;
    delete address.stateId;

    return this.prisma.user.create({
      data: {
        ...user,
        ...(cityId && {
          address: {
            create: {
              ...address,
              city: {
                connect: {
                  id: cityId,
                },
              },
              state: {
                connect: {
                  id: stateId,
                },
              },
            },
          },
        }),
      },
    });
  }

**Comment by ryands17** [comment, marked as answer]
Hey @jeancarlos-cpu :wave: 
One way to do this would be to loop over the optional properties and add them to the data object.

```js
const { cityId, stateId } = address;
delete address.cityId;
delete address.stateId;

const data = {};
for (let item in address) {
  if (item) data[item] = address[item]
}

prisma.user.create({
  data
});
```

**Comment by ludralph** [comment]
Hi @jeancarlos-cpu,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
